testsupport/tiers/nstemplateset.go

```diff
@@ -35,6 +39,30 @@ func VerifyNSTemplateSet(t *testing.T, hostAwait *wait.HostAwaitility, memberAwa
 				checkNamespaceObjects(t, ns, memberAwait, nsTmplSet.Name)
 			}(check)
 		}
+		spaceRoles := map[*templatev1.Template][]string{}
+		decoder := serializer.NewCodecFactory(hostAwait.Client.Scheme()).UniversalDeserializer()
+		for _, r := range nsTmplSet.Spec.SpaceRoles {
+			tmpl, err := hostAwait.WaitForTierTemplate(r.TemplateRef)
+			require.NoError(t, err)
+			t.Logf("space role template: %s / %s", tmpl.GetName(), tmpl.Spec.Template.GetName())
+			for i, rawObj := range tmpl.Spec.Template.Objects {
+				var obj unstructured.Unstructured
+				_, _, err := decoder.Decode(rawObj.Raw, nil, &obj)
+				require.NoError(t, err)
+				tmpl.Spec.Template.Objects[i].Object = &obj
+			}
```

Comment:
removed in done in [32e89ef](https://github.com/codeready-toolchain/toolchain-e2e/pull/518/commits/32e89ef09f6a29ae7d457910f467be60298b5eb1) and [c6c87de](https://github.com/codeready-toolchain/toolchain-e2e/pull/518/commits/c6c87de23df3b851b66ea81efc0d0b1eb736a3e5)

Reply:
restored in 73c46c2, because in the `TestUpdateNSTemplateTier` test, we have tier templates with names that don't match the prefix, but we can work-around this by checking the name of their embedded OpenShift template. 
An alternative would be to add more custom prefix matches in the check logic, but I'm not too fond about this solution :/

Reply:
> because in the TestUpdateNSTemplateTier test, we have tier templates with names that don't match the prefix

ah, I see - it's because we create brand new TierTemplates that have different name format. However, they should still have the correct label `type` (admin,view,...) so you could use that.

>  but we can work-around this by checking the name of their embedded OpenShift template.

but why do you need to decode all inner objects? The name of the OpenShift template should be enough, right?

Reply:
> ah, I see - it's because we create brand new TierTemplates that have different name format. However, they should still have the correct label type (admin,view,...) so you could use that.

good point, let me see if I can use the labels instead, then

> but why do you need to decode all inner objects? The name of the OpenShift template should be enough, right?

hmm, right, I can probably skip that part. Let me see

Reply:
>> ah, I see - it's because we create brand new TierTemplates that have different name format. However, they should still have the correct label type (admin,view,...) so you could use that.

> good point, let me see if I can use the labels instead, then

@MatousJobanek AFAICS, TierTemplates don't have such labels :/

Reply:
> but why do you need to decode all inner objects? The name of the OpenShift template should be enough, right?

done in 65ba041

Reply:
ah, right - for some reason I thought that it's a label - confused it with namespaces. It's actually stored in the spec https://github.com/codeready-toolchain/api/blob/master/api/v1alpha1/tiertemplate_types.go#L25

Reply:
done in f8ab1be